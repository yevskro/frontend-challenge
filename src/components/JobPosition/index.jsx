import React from 'react';
import styled from 'styled-components';
import JobInfo from '../JobInfo';
import JobCompany from '../JobCompany';

function JobPosition({ children, postedAt, contract, location, company }) {
  return (
    <StyledJobPosition>
      <JobCompany newly={company.newly} featured={company.featured}>
        {company.name}
      </JobCompany>
      <Heading>{children}</Heading>
      <JobInfo postedAt={postedAt} contract={contract} location={location} />
    </StyledJobPosition>
  );
}

const Heading = styled.h3`
  padding-top: 11px;
  padding-bottom: 11px;
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  margin: 0px;
  white-space: nowrap;
  &:hover {
    color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
    cursor: pointer;
  }
`;

const StyledJobPosition = styled.div`
  display: flex;
  flex-direction: column;
  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.medium};
  }
  @media (min-width: 600px) {
    font-size: ${({ theme }) => theme.fontSize.large};
  }
`;

export default JobPosition;
