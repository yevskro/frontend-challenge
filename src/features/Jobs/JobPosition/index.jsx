import React from 'react';
import styled from 'styled-components';
import JobInfo from '../JobInfo';
import JobCompany from '../JobCompany';

function JobPosition({ children, postedAt, contract, location, company }) {
  return (
    <StyledJobPosition>
      <JobCompany newly={company.newly} featured={company.featured}>
        {company.name}
      </JobCompany>
      <Heading>{children}</Heading>
      <JobInfo postedAt={postedAt} contract={contract} location={location} />
    </StyledJobPosition>
  );
}

const Heading = styled.h3`
  padding: 11px 0px 11px 0px;
  margin: 0px;
  white-space: nowrap;
  font-weight: ${({ theme }) => theme.fontWeight.bold};

  &:hover {
    color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
    cursor: pointer;
  }
`;

const StyledJobPosition = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.medium};
  }

  @media (min-width: 600px) {
    font-size: ${({ theme }) => theme.fontSize.large};
  }
`;

export default JobPosition;
