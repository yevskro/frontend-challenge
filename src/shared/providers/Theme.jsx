import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
