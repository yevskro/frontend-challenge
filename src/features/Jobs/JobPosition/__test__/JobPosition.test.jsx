import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Theme from '../../../../shared/providers/Theme';
import JobPosition from '..';

test('renders', () => {
  render(
    <Theme>
      <JobPosition
        company={{
          name: 'company name',
          newly: true,
          featured: true,
        }}
        postedAt="posted at"
        contract="contract"
        location="location"
      >
        position
      </JobPosition>
    </Theme>
  );

  expect(screen.getByText('company name')).toBeInTheDocument();
  expect(screen.getByText('NEW!')).toBeInTheDocument();
  expect(screen.getByText('FEATURED')).toBeInTheDocument();
  expect(screen.getByText('posted at')).toBeInTheDocument();
  expect(screen.getByText('contract')).toBeInTheDocument();
  expect(screen.getByText('location')).toBeInTheDocument();
  expect(screen.getByText('position')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <JobPosition
          company={{
            name: 'company name',
            newly: true,
            featured: true,
          }}
          postedAt="posted at"
          contract="contract"
          location="location"
        >
          position
        </JobPosition>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
