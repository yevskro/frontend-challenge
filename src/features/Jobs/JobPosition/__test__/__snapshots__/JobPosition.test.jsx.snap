// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c7 {
  white-space: nowrap;
}

.c7:first-child::before {
  font-size: 1.7rem;
  content: ' ';
}

.c7:not(:first-child)::before {
  display: inline-block;
  content: '.';
  padding: 0px 4px 0px 4px;
  font-size: 1.7rem;
  -webkit-transform: translateY(-16%);
  -ms-transform: translateY(-16%);
  transform: translateY(-16%);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: hsl(180,8%,52%);
  font-weight: 400;
  font-size: 0.9rem;
}

.c2 {
  height: 24px;
  line-height: 27px;
  vertical-align: middle;
  padding: 0px 9px 0px 9px;
  margin: 0px 4px 0px 4px;
  color: white;
  border-radius: 20px;
  font-size: 0.9rem;
}

.c3 {
  margin-left: 18px;
  background-color: hsl(180,29%,50%);
  font-size: 0.7rem;
}

.c4 {
  background-color: hsl(180,14%,20%);
  font-size: 0.7rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  color: hsl(180,29%,50%);
  font-weight: 600;
  font-size: 0.8rem;
}

.c5 {
  padding: 11px 0px 11px 0px;
  margin: 0px;
  white-space: nowrap;
  font-weight: 600;
}

.c5:hover {
  color: hsl(180,29%,50%);
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:600px) {
  .c7:not(:first-child)::before {
    padding: 0px 8px 0px 8px;
  }
}

@media (min-width:600px) {
  .c7 {
    font-size: 1rem;
  }
}

@media (min-width:500px) {
  .c1 {
    font-size: 0.9rem;
  }
}

@media (min-width:500px) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:600px) {
  .c0 {
    font-size: 1.2rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h6>
      company name
    </h6>
    <span
      className="c2 c3"
    >
      NEW!
    </span>
    <span
      className="c2 c4"
    >
      FEATURED
    </span>
  </div>
  <h3
    className="c5"
  >
    position
  </h3>
   
  <ul
    className="c6"
  >
    <li
      className="c7"
    >
      posted at
    </li>
    <li
      className="c7"
    >
      contract
    </li>
    <li
      className="c7"
    >
      location
    </li>
  </ul>
</div>
`;
